pennylane/qchem/structure.py

```diff
@@ -282,7 +282,32 @@ def excitations(electrons, orbitals, delta_sz=0):
     return singles, doubles
 
 
-def hf_state(electrons, orbitals):
+def _beta_matrix(orbitals):
+    r"""Generate the beta matrix for conversion from occupation number basis to the Bravyi-Kitaev basis.
+
+    Args:
+        orbitals (int): Number of *spin* orbitals. If an active space is defined,
+            this is the number of active spin-orbitals.
+
+    Returns:
+        (array): Transformation matrix beta
+    """
+
+    # Base case for beta_1
+    if orbitals == 1:
+        return np.array([[1]])
+
+    bin_range = int(np.ceil(np.log2(orbitals)))
+
+    beta_previous = _beta_matrix(2 ** (bin_range - 1))
+    beta = np.kron(np.eye(2), beta_previous)
+    # Fill in the last row of the last quadrant with 1's
+    beta[-1, : 2 ** (bin_range - 1)] = 1
+
+    return beta[:orbitals, :orbitals]
```

Comment:
Just a comment. I think it's interesting that your base case here is an np.array([1]) but the function itself takes in an integer not an array. It can be a little confusing because the recursion relies on this idea that _beta_matrix(np.array([1])) is equivalent to _beta_matrix(1). There's like some automatic conversion here where the arrays are converted to Integers. Also something to note is that if you have like 100000 orbitals this would probably crash your kernel due to the recursion. I don't see an obvious way to improve the efficiency though.

Reply:
`def _beta_matrix(orbitals):
    if orbitals == 1:
        return np.array([[1]])

    bin_range = int(np.ceil(np.log2(orbitals)))

    beta = np.array([[1]])

    for i in range(bin_range):
        beta = np.kron(np.eye(2), beta)
        beta[-1, :2**i] = 1

    return beta[:orbitals, :orbitals]`

Reply:
+1 for not using recursion unless there is no better way.